crypto/pkcs12/p12_crpt.c

```diff
@@ -54,15 +55,19 @@ int PKCS12_PBE_keyivgen_ex(EVP_CIPHER_CTX *ctx, const char *pass, int passlen,
         PBEPARAM_free(pbe);
         return 0;
     }
-    if (!PKCS12_key_gen_utf8_ex(pass, passlen, salt, saltlen, PKCS12_IV_ID,
-                                iter, EVP_CIPHER_iv_length(cipher), iv, md,
-                                libctx, propq)) {
-        ERR_raise(ERR_LIB_PKCS12, PKCS12_R_IV_GEN_ERROR);
-        PBEPARAM_free(pbe);
-        return 0;
+    if (EVP_CIPHER_iv_length(cipher) > 0) {
```

Comment:
This looks like a change in behaviour?.. Before it would have raised an error I assume.. Is that what you want?

Reply:
Yes, this is the fix for PBEs using ciphers which do not specify an IV, e.g. RC4. I guess they weren't tested before?